lib/dynflow/rails/daemon.rb

```diff
@@ -1,32 +1,54 @@
 require 'fileutils'
+require 'daemons'
+require 'get_process_mem'
+require 'dynflow/watchers/memory_consumption_watcher'
 
 module Dynflow
   class Rails
     class Daemon
+      attr_reader :dynflow_memory_watcher_class, :daemons_class
+
+      # make Daemon dependency injection ready for testing purposes
+      def initialize(
+        dynflow_memory_watcher_class = ::Dynflow::Watchers::MemoryConsumptionWatcher,
+        daemons_class = ::Daemons
+      )
+        @dynflow_memory_watcher_class = dynflow_memory_watcher_class
+        @daemons_class = daemons_class
+      end
+
       # Load the Rails environment and initialize the executor in this thread.
-      def run(rails_root = Dir.pwd)
+      def run(rails_root = Dir.pwd, options = {})
         STDOUT.puts('Starting Rails environment')
         rails_env_file = File.expand_path('./config/environment.rb', rails_root)
         unless File.exist?(rails_env_file)
           raise "#{rails_root} doesn't seem to be a Rails root directory"
         end
+
+        STDERR.puts("Starting dynflow with the following options: #{options}")
+
         ::Rails.application.dynflow.executor!
+
+        if options[:memory_limit] > 0
+          ::Rails.application.dynflow.config.on_init do |world|
```

Comment:
It's dependent on -tasks already - https://github.com/theforeman/foreman-tasks/commit/d3051b91d63a1c809b78fbd48187ba364c391a89#diff-1f87f95c00892140947fc8f17d3f8980R34 , this module just tries to find the world used by the Rails application